examples/dart/test/open_flexible_sync_realm_test.dart

```diff
@@ -18,28 +16,102 @@ class _Tricycle {
 void main() {
   group('Open Flexible Sync Realm', () {
     const APP_ID = "flex-config-tester-vwevn";
-    AppConfiguration appConfig = AppConfiguration(APP_ID);
-    App app = App(appConfig);
+    final appConfig = AppConfiguration(APP_ID);
+    final app = App(appConfig);
     test("Open Flexible Sync Realm", () async {
-      Credentials credentials = Credentials.anonymous();
+      final credentials = Credentials.anonymous();
       // :snippet-start: open-flexible-sync-realm
-      User currentUser = await app.logIn(credentials);
-      Configuration config = Configuration.flexibleSync(
-          currentUser, [Tricycle.schema],
+      final currentUser = await app.logIn(credentials);
+      final config = Configuration.flexibleSync(currentUser, [Tricycle.schema],
           path: 'flex.realm');
-      Realm realm = Realm(config);
+      final realm = Realm(config);
       // :snippet-end:
       expect(realm.isClosed, false);
       expect(app.currentUser?.id != null, true);
       await cleanUpRealm(realm, app);
       expect(realm.isClosed, true);
       expect(app.currentUser, null);
     });
+    test('Async Open Flexible Sync Realm', () async {
+      Credentials credentials = Credentials.anonymous();
+      User currentUser = await app.logIn(credentials);
+      // :snippet-start: async-open
+      // Helper function to check if device is connected to the internet.
+      Future<bool> isDeviceOnline() async {
+        // ...logic to check if device is online
+        return true; // :remove:
+      }
 
-    test("Handle Sync Error", () async {
-      var handlerCalled = false;
+      final config = Configuration.flexibleSync(currentUser, [Tricycle.schema]);
+      // Only use asynchronous open if app is online.
+      late Realm realm;
+      if (await isDeviceOnline()) {
+        realm = await Realm.open(config);
```

Comment:
This is super subjective, but I wonder if we might want to put a comment in here and above ln 51 explicitly stating what we're doing? If people copy-paste this without really reading the docs & understanding what these APIs do, they may not realize exactly what this logic does. i.e.:

```suggestion
        // If the device is online, download changes and then open the realm
        realm = await Realm.open(config);
```

And then down below, something like:

```
} else {
   // If the device is offline, open a local realm
   realm = Realm(config);
```